pkg/bench/rttanalysis/validate_benchmark_data.go

```diff
@@ -58,193 +60,152 @@ var (
 
 // RunBenchmarkExpectationTests runs tests to validate or rewrite the contents
 // of the benchmark expectations file.
-func RunBenchmarkExpectationTests(t *testing.T) {
-	expectations := readExpectationsFile(t)
-
-	benchmarks := getBenchmarks(t)
-	if *rewriteFlag != "" {
-		rewriteBenchmarkExpectations(t, benchmarks)
+func runBenchmarkExpectationTests(t *testing.T, r *Registry) {
+	if util.IsMetamorphicBuild() {
+		execTestSubprocess(t)
 		return
 	}
 
+	defer log.Scope(t).Close(t)
```

Comment:
i wonder if this belongs in `TestBenchmarkExpectations` at the top level, but not fussed. i  guess would it be worth including `//go:generate ../../util/leaktest/add-leaktest.sh *_test.go` somewhere would have automatically added this.